spec/21-http_header_format_rationale.md

```diff
@@ -98,17 +90,103 @@ configuration - certain header mutations may still be required. An example may b
 smart sampling mechanisms that rely on additional data propagation in
 `tracestate`.
 
-### Size limits
+## Size limits of `tracestate`
+
+### Total size limit
+
+The field `tracestate` opens up rich interoperability and extensibility
+scenarios for vendors. There are defined use cases for the `tracestate`
+like positioning request in multiple distributed tracing graphs and
+providing backward compatibility with older protocols. But `tracestate`
+is not limited to this use and will drive innovations and new scenarios
+going forward.
+
+The opaque nature of `tracestate` introduces some tension between
+guarantees to vendors around propagating vendor-specific data with
+traces vs. storage and propagation constraints.
+
+On the one hand, field value should be small so implementors can satisfy
+the requirement to pass the value all the time. It is especially
+critical for various messaging systems where metadata like `tracestate`
+is not paid for by the customer and provided for free.
+
+On the other hand, in order for the field to be useful there should be some
+guarantees that it will in fact be propagated in most of the cases.
+
+Without changing definition of `tracestate` to make it less opaque, any
+declared limit will be arbitrary. It may be based on discussions with
+vendors on their needs. But it's still arbitrary as - first, it may not
+account for the needs of everybody and second - whatever guarantee will
+be provided - it will be abused as guaranteed propagation is a very
+tempting and hard to implement feature.
+
+Talking about abuse - one thing this working group is doing is working
+on user-defined context propagation as part of [Correlation
+Context](https://w3c.github.io/correlation-context/) spec. Correlation
+context will provide a relieve valve for people who want to have freedom
+propagating relatively big payload across the components of a
+distributed trace.
+
+This specification allows removing and adding as many `tracestate`
+entries as an implementation needs. This freedom is required to satisfy
+privacy and interoperability concerns. Thus all suggestions
+specification is making regarding the size of `tracestate` field will
+only be a recommendation that would improve vendors interoperability and
+cannot be "enforced" in practice.
+
+Possible solutions:
```

Comment:
Are these "considered and rejected"? The caption is confusing.

Reply:
I addressed feedback. Also just a note, it's in rationale doc, not a spec